### PR TITLE
Adding Uploading Capability to 526 pdf

### PR DESCRIPTION
**Pull request by bastosmichael**
## Description of change
Adding document functionality to 526 for ticket https://github.com/department-of-veterans-affairs/vets-contrib/issues/3435

## Testing done
- Rspec testing

## Testing planned
- Will test in dev-api.va.gov

## Acceptance Criteria (Definition of Done)
- Create the ability to upload a specific 526 form through a PUT end point.

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/claims_api/app/models/claims_api/auto_established_claim.rb [hunk 1]
```diff
@@ -64,5 +65,23 @@ def set_md5
     def status_from_phase(*)
       status
     end
+
+    def file_name
```
Comment (ATeal) [inline modules/claims_api/app/models/claims_api/auto_established_claim.rb]: I feel like these same methods are duplicated all over the place in claims api module...could you refactor?

File: Gemfile.lock [hunk 2]
```diff
@@ -501,7 +501,8 @@ GEM
     rack (2.0.7)
     rack-attack (5.0.1)
       rack
-    rack-cors (0.4.1)
+    rack-cors (1.0.6)
```
Comment (LindseySaari) [inline Gemfile.lock]: Wondering if update for rack and nokogiri should be in a separate PR?

Comment (bastosmichael) [reply Gemfile.lock]: There is a separate PR for the gemfile updates but I also included them in here because they were causing a CI/CD failure for this PR https://github.com/department-of-veterans-affairs/vets-api/pull/3560/files

File: modules/claims_api/app/models/claims_api/concerns/file_data_validation.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  module FileDataValidation
```
Comment (ATeal) [inline modules/claims_api/app/models/claims_api/concerns/file_data_validation.rb]: this doesn't actually validate anything so maybe a different name?

File: modules/claims_api/app/models/claims_api/auto_established_claim.rb (hunk 1 above)
Comment (bastosmichael) [reply modules/claims_api/app/models/claims_api/auto_established_claim.rb]: Refactored.

File: modules/claims_api/app/models/claims_api/concerns/file_data_validation.rb (hunk 3 above)
Comment (bastosmichael) [reply modules/claims_api/app/models/claims_api/concerns/file_data_validation.rb]: Just calling it FileData then...
